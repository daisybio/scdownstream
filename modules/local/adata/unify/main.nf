process ADATA_UNIFY {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'oras://community.wave.seqera.io/library/mygene_scanpy:fe39686ba0c901e7':
        'community.wave.seqera.io/library/mygene_scanpy:b8864cab99d32be2' }"

    input:
    tuple val(meta), path(h5ad)

    output:
    tuple val(meta), path("*.h5ad"), emit: h5ad
    path "versions.yml"            , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    prefix = task.ext.prefix ?: "${meta.id}"
    unify_gene_symbols = task.ext.unify_gene_symbols ?: false
    template 'unify.py'
}
