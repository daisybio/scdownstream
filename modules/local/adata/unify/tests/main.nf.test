nextflow_process {
    name 'Test Process ADATA_UNIFY'
    script '../main.nf'
    process 'ADATA_UNIFY'

    tag "modules"
    tag "adata/unify"
    tag "adata"

    test("test_adata_unify") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file("https://raw.githubusercontent.com/nf-core/test-datasets/scdownstream/samples/SAMN14430799_custom_emptydrops_filter_matrix.h5ad", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll(
                {assert process.success},
                {assert snapshot(process.out.h5ad).match("adata_unify_h5ad")},
                {assert snapshot(process.out.versions).match("adata_unify_versions")}
            )
        }
    }

    test("test_adata_unify - stub") {
        options '-stub'
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file("https://raw.githubusercontent.com/nf-core/test-datasets/scdownstream/samples/SAMN14430799_custom_emptydrops_filter_matrix.h5ad", checkIfExists: true),
                ]
                """
            }
        }
        then {
            assertAll(
                {assert process.success},
                {assert snapshot(process.out.h5ad).match("adata_unify_h5ad_stub")},
                {assert snapshot(process.out.versions).match("adata_unify_versions_stub")}
            )
        }
    }
}
