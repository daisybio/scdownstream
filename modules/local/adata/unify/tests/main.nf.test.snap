{
    "adata_unify_h5ad": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test_null_unified.h5ad:md5,d5606e8cc7cacce7a09b7b7a75436f2f"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-17T14:45:51.190945122"
    },
    "adata_unify_h5ad_stub": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test_null_unified.h5ad:md5,d5606e8cc7cacce7a09b7b7a75436f2f"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-17T14:46:32.93427331"
    },
    "adata_unify_versions_stub": {
        "content": [
            [
                "versions.yml:md5,d3a2dda51a44940594c4a71d5d41fd27"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-17T14:46:33.579818268"
    },
    "adata_unify_versions": {
        "content": [
            [
                "versions.yml:md5,d3a2dda51a44940594c4a71d5d41fd27"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-17T14:45:51.646132076"
    }
}